### PR TITLE
feat: cache get_network_difficulty response

### PR DESCRIPTION
**Pull request by mmrrnn**
Cache `get_network_difficulty`  response since it's too heavy to call it every 1sec

### CODE REVIEW

**Comment by brianp** [comment]
Should it be noticeable via the logs that we receive less timeout, or already fetching notifications?

**Comment by mmrrnn** [comment]
I tested it with timer locally and I went from like 30/40ms to 2/3ms -> Not a big deal but maybe in some cases it's more significant. This is called once per second.

File: src-tauri/src/node_manager.rs [hunk 1]
```diff
@@ -171,10 +171,10 @@ impl NodeManager {
     pub async fn get_network_hash_rate_and_block_reward(
         &self,
     ) -> Result<(u64, u64, MicroMinotari, u64, u64, bool), NodeManagerError> {
-        let status_monitor_lock = self.watcher.read().await;
+        let mut status_monitor_lock = self.watcher.write().await;
         let status_monitor = status_monitor_lock
             .status_monitor
-            .as_ref()
+            .as_mut()
```
Comment (brianp) [inline src-tauri/src/node_manager.rs:177]: Why did this need to become mutable?

Comment (mmrrnn) [reply src-tauri/src/node_manager.rs:177]: We save cached properties

**Review by brianp — APPROVED** [review]
Tested, no new issues found. Seems to be working.

**Pull request merged**